contract/contracts/EventManager.sol

```diff
@@ -24,14 +22,16 @@ contract EventManager {
         owner.transfer(address(this).balance);
     }
 
-    function createEvent(bytes calldata cid, uint16 ticketsTotal, uint256 endDate) public payable {
+    function createEvent(string calldata eventName, string calldata imageURI, bytes calldata cid, uint16 ticketsTotal, uint256 endDate) public payable {
         require(msg.value >= fee, "too small fee");
-        Event memory newEvent;
-        newEvent.cid = cid;
-        newEvent.ticketsTotal = ticketsTotal;
-        newEvent.endDate = endDate;
-        events[msg.sender].push(newEvent);
+        EventNFT newEvent = new EventNFT(msg.sender, eventName, string.concat("EVNT", Strings.toString(createdEvents.length + 1)), imageURI, ticketsTotal, cid, endDate);
```

Comment:
do we really need `string.concat("EVNT", Strings.toString(createdEvents.length + 1))`?
maybe it would be better to have prepopulated field on FE that the user could edit?

Reply:
will improve it with the next PR